src/EditorFeatures/CSharpTest/EditAndContinue/StatementEditingTests.cs

```diff
@@ -10553,5 +10553,67 @@ public void WithExpression_PropertyValueReorder()
         }
 
         #endregion
+
+        #region Top Level Statements
+
+        [Fact]
+        public void TopLevelStatement_CaptureArgs()
+        {
+            var src1 = @"
+using System;
+
+var x = new Func<string>(() => ""Hello"");
+
+Console.WriteLine(x());
+";
+            var src2 = @"
+using System;
+
+var x = new Func<string>(() => ""Hello"" + args[0]);
+
+Console.WriteLine(x());
+";
+            var edits = GetTopEdits(src1, src2);
+
+            // TODO: allow creating a new leaf closure
+            edits.VerifySemanticDiagnostics(
+                Diagnostic(RudeEditKind.CapturingVariable, "using System;\r\n\r\nvar x = new Func<string>(() => \"Hello\" + args[0]);\r\n\r\nConsole.WriteLine(x());\r\n", "args"));
```

Comment:
>Diagnostic [](http://example.com/codeflow?start=16&length=10)

Seems like the diagnostic span is not idea - we shouldn't squiggle the whole file.

Reply:
There isn't much else to go on. Could just squiggle the first global statement?

The capture info only returns the parameter info for the `args` parameter, and the "member body" in question is the compilation unit, so can't narrow that down any. Unfortunately the code flow analysis doesn't have any capability for knowing _where_ something was captured.

Reply:
Hmm, let's leave it as is for now but file an issue to follow up. Maybe a compiler API is needed that tells us where the variables are captured (the lambda or the variable usage).

Reply:
https://github.com/dotnet/roslyn/issues/54370